### PR TITLE
Add safety page for ardupilot

### PR DESCRIPTION
**Pull request by hamishwillee**
Seems good time to do this. Not all the ArduPilot docs are clear on failsafe options.

### CODE REVIEW

File: en/SetupView/safety_ardupilot.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+- **Critical action** ([BATT_FS_CRT_ACT](http://ardupilot.org/copter/docs/parameters.html#batt-fs-crt-act-critical-battery-failsafe-action)) - Select one of: None, Land, RTL, SmartRTL, SmartRTL or Land, Terminate.
+- **Low voltage threshold** ([BATT_LOW_VOLT](http://ardupilot.org/copter/docs/parameters.html#batt-low-volt-low-battery-voltage)) - Battery voltage that triggers the *low action*. 
+- **Critical voltage threshold** ([BATT_CRT_VOLT](http://ardupilot.org/copter/docs/parameters.html#batt-crt-volt-critical-battery-voltage))- Battery voltage that triggers the *critical action*. 
+- **Low mAh threshold** ([BATT_LOW_MAH](http://ardupilot.org/copter/docs/parameters.html#batt-low-mah-low-battery-capacity)) - Battery capacity that triggers the *low action*. 
+- **Critical mAh threshold** ([BATT_CRT_MAH](http://ardupilot.org/copter/docs/parameters.html#batt-crt-mah-battery-critical-capacity)) - Battery capacity that triggers the *critical action*. 
+
+
+<!-- Questions
+- What about Battery 2?
```
Comment (hamishwillee) [inline en/SetupView/safety_ardupilot.md]: There appear to be Battery 2 items in the parameters for copter. How are these supported? (ie if I have two batteries do I end up with two panels, or more sensibly, a combined failsafe?

File: en/SetupView/safety_ardupilot.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+- What actions do BATT_LOW_MAH (capacity) trigger. The ones listed in params don't exist. Is it just same as voltage ones?)
```
Comment (hamishwillee) [inline en/SetupView/safety_ardupilot.md]: Ignore this comment. The answer is yes.

File: en/SetupView/safety_ardupilot.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+![Safety Setup - General Failsafe Triggers (Copter)](../../assets/setup/safety/safety_arducopter_general_failsafe_triggers.jpg)
+
+The configuration options are:
+- **Ground Station failsafe** - Disabled, Enabled always RTL, Enabled Continue with Mission in Auto Mode, Enabled Always SmartRTL or RTL, Enabled Always SmartRTL or Land.
+- **Throttle failsafe** - Disabled, Always RTL, Continue with Mission in Auto Mode, Always land.
+- **PWM Threshold** ([FS_THR_VALUE](http://ardupilot.org/copter/docs/parameters.html#fs-thr-value-throttle-failsafe-value)) - PWM value below which throttle failsafe triggers.
+
+<!-- Question 
+- no radio failsafe?
```
Comment (hamishwillee) [inline en/SetupView/safety_ardupilot.md]: Why no radio link loss failsafe?

File: en/SetupView/safety_ardupilot.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+### Geofence {#geofence_copter}
+
+This panel sets the parameters for the cylindrical [Simple Geofence](http://ardupilot.org/copter/docs/ac2_simple_geofence.html). 
+You can set whether the fence radius or height are enabled, the maximum values for causing a breach, and the action in the event of a breach.
+
+![Safety Setup - Geofence (Copter)](../../assets/setup/safety/safety_arducopter_geofence.jpg)
+
+The configuration options are:
+
+- **Circle GeoFence enabled** () - Enable the circular geofence.
```
Comment (hamishwillee) [inline en/SetupView/safety_ardupilot.md]: How does this work for enabling? Doesn't seem to be corresponding parameter.

File: en/SetupView/safety_ardupilot.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+### Failsafe Actions {#failsafe_actions_sub}
+
+The configuration options are:
+
+- **GCS Heartbeat** - Select one of: Disabled, Warn only, Disarm, Enter depth hold mode, Enter surface mode.
+- **Leak** - Select one of: Disabled, Warn only, Enter surface mode.
+  - **Detector Pin** - Select one of: Disabled, Pixhawk Aux (1-6), Pixhawk 3.3ADC(1-2), Pixhawk 6.6ADC.
+  - **Logic when Dry** - Select one of: Low, High.
+- **Battery** - ?.
```
Comment (hamishwillee) [inline en/SetupView/safety_ardupilot.md]: Battery selector empty in Sub. Is that OK?

File: en/SetupView/safety_ardupilot.md (hunk 4 above)
Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: Circle and Altitude GeoFence are controlled by the FENCE_ENABLE parameter which these checkboxes set appropriately.

File: en/SetupView/safety_ardupilot.md (hunk 5 above)
Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: I don't know anything about Sub. You'll have to ask the Sub folks. @jaxxzer?

File: en/SetupView/safety_ardupilot.md (hunk 1 above)
Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: You end up with two panels. In order to get the second panel to show you need to enable the second batter on the Power page.

File: en/SetupView/safety_ardupilot.md (hunk 3 above)
Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: Hmm, good question. I don't think that was every implemented. What's the parameter for that?

File: en/SetupView/safety_ardupilot.md (hunk 1 above)
Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: Thanks, I'll tidy.

@rmackay9 - FYI, IMO doing safety settings individually on multiple batteries seems a little error prone- easy to end up getting low battery warnings when you have plenty of power in one battery but not the other.

File: en/SetupView/safety_ardupilot.md (hunk 3 above)
Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: I think I was wrong for [radio failsafe](http://ardupilot.org/copter/docs/radio-failsafe.html) - it exists but has no parameter setup (depends on a bunch of other conditions).

There may be some [others missing](http://ardupilot.org/copter/docs/failsafe-landing-page.html)  though: ekf/dcm, polygon fence, crash check, parachute, ADSB.

File: en/SetupView/safety_ardupilot.md (hunk 4 above)
Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: Ah, I think you mean FENCE_TYPE (+ FENCE_ENABLE). I'll update appropriately.

File: en/SetupView/safety_ardupilot.md (hunk 5 above)
Comment (jaxxzer) [reply en/SetupView/safety_ardupilot.md]: I'll tune in today and take a look at this.

Comment (jaxxzer) [reply en/SetupView/safety_ardupilot.md]: @hamishwillee can you show me what you are seeing? Our failsafes (on stable firmware) are documented here: https://www.ardusub.com/operators-manual/failsafes.html

Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: @jaxxzer This is recent (in the last week) master, on Ubuntu 16.04 using QGC 3.4.3.

Battery field is empty, and dropdown is empty:
image

Might be related to
image

File: en/SetupView/safety_ardupilot.md (hunk 3 above)
Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: There are lot's of things you can futz with paramters. Only the most important once which are used by most users are exposed in the setup pages ui. The higher end, less used, power user stuff is just left to the parameter editor.

QGC does not support ArduPilot polygon fences or rally points. Long story on that one I'd rather no go into.

Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: Makes sense. I will add a note up the top explaining this. You otherwise OK to merge?

Comment (DonLakeFlyer) [reply en/SetupView/safety_ardupilot.md]: > You otherwise OK to merge?

yup

**Comment by hamishwillee** [comment]
FYI, In parent safety page I just added links to this new doc and to PX4 safety docs. I intend to also provide a specific PX4 safety page, so this is a placeholder action.

File: en/SetupView/safety_ardupilot.md (hunk 5 above)
Comment (jaxxzer) [reply en/SetupView/safety_ardupilot.md]: Ok. It appears that you are on v3.6.0-dev of ArduSub, which qgc does not support at the moment. We are working on bringing things up to speed for ArduSub on QGC master here.

Comment (hamishwillee) [reply en/SetupView/safety_ardupilot.md]: Thanks!

**Pull request merged**